test/test_discriminator.py

```diff
@@ -0,0 +1,129 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test version string generation."""
+
+import numpy as np
+from qiskit.providers import BaseBackend
+from qiskit.providers.models import QasmBackendConfiguration
+from qiskit.result import Result
+from qiskit_experiments import ExperimentData
+from qiskit.test import QiskitTestCase
+
+
+from qiskit_experiments.measurement.discriminator import (
+    DiscriminatorExperiment,
+    DiscriminatorAnalysis,
+)
+
+
+class DiscriminatorBackend(BaseBackend):
```

Comment:
I suggest using the `MockIQBackend` which will allow you to test the discriminator and minimize the amount of code needed. Note that this is currently being updated in https://github.com/Qiskit/qiskit-experiments/pull/74